libraries/freertos_plus/standard/https/src/iot_https_client.c

```diff
@@ -800,6 +844,53 @@ static int _httpParserOnChunkCompleteCallback(http_parser * pHttpParser)
 
 /*-----------------------------------------------------------*/
 
+static IotHttpsReturnCode_t _receiveHttpsBodyAsync(_httpsResponse_t* pHttpsResponse)
+{
+    HTTPS_FUNCTION_ENTRY(IOT_HTTPS_OK);
+    /* Get the request reference from the response for the callback function pointers. */
+    _httpsRequest_t* pHttpsRequest = pHttpsResponse->pHttpsRequest;
+
+    if(pHttpsRequest->pCallbacks->readReadyCallback)
+    {
+        /* If there is still more body that has not been passed back to the user, then this callback is invoked again. */
+        do {
+            pHttpsRequest->pCallbacks->readReadyCallback(pHttpsRequest->pUserPrivData, 
+                pHttpsResponse, 
+                pHttpsResponse->bodyRxStatus, 
+                pHttpsResponse->status);
+            if(pHttpsResponse->cancelled == true)
+            {
+                IotLogDebug("Cancelled HTTP response %d.", pHttpsResponse);
+                status = IOT_HTTPS_ASYNC_CANCELLED;
+                /* We break out of the loop and do not goto clean up because we want to print debugging logs for 
+                   the parser state and the networks status. */
+                break;
+            }
+        } while((pHttpsResponse->parserState < PARSER_STATE_BODY_COMPLETE) && (pHttpsResponse->bodyRxStatus == IOT_HTTPS_OK));
```

Comment:
who updates ```pHttpsResponse->parserState``` or ```pHttpsResponse->bodyRxStatus == IOT_HTTPS_OK```. Are these updated from a different thread? Is there a thread synchronization needed?

Reply:
pHttpsResponse->parserState is updated in only in the http-parser callbacks implemented. pHttpsResponse->bodyRxStatus is updated during the readReadyCallback() (implemented by the application) when the network data is read. These are all from the same context.